.chloggen/ottl-communityid-function.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'enhancement'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/ottl
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Introduce `CommunityID` function to generate network hash (20-byte SHA1 digest) flow from the given source and destination {IP, port}, optionally protocol and seed values.
```

Comment:
Can you explain why this function is being added and when it would be useful?

Reply:
Hi @atoulme,
It is a standardized mechanism, useful when to track/monitor the network behaviour in security (or o11y or NSM) applications. Here some discussions under the attached issue - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34062#issuecomment-2256759340
Another visual content for understanding it easier: https://suricon.net/wp-content/uploads/2019/01/SuriCon2018_Kreibich.pdf